### PR TITLE
[ADD] load resources using external function by id

### PR DESCRIPTION
**Pull request by archetipo**

### CODE REVIEW

**Comment by bobslee** [comment]
Hi @archetipo , I tried to get the idea (aim) of this feature.
I'm still clueless after checking the code, 
Please could you describe the usage (aim)? maybe add a short description or example in the README?
Thanks!

File: formiodata/builder.py [hunk 1]
```diff
@@ -25,6 +25,7 @@ def __init__(self, schema_json, **kwargs):
         # i18n (translations)
         self.i18n = kwargs.get('i18n', {})
         self.resources = kwargs.get('resources', {})
+        self.resources_ext = kwargs.get('resources_ext', False)
```
Comment (bobslee) [inline formiodata/builder.py:28]: I tried to get the idea (aim) of this feature.
I'm still clueless after checking the code.
Please could you describe the usage (aim)? maybe add a short description or example in the README?
Thanks!

**Review by bobslee — CHANGES_REQUESTED** [review]
I tried to get the idea (aim) of this feature.
I'm still clueless after checking the code.
Please could you describe the usage (aim)? maybe add a short description or example in the README?
Thanks!

**Comment by archetipo** [comment]
Hi @bobslee,
 
 Resources are custom data referenced in field, that you can use to fill options in a select or resource field.

Look test case [here](https://github.com/novacode-nl/python-formio-data/pull/4/commits/84c64223d3bbdf894e305d1d956d6a36f885ff04#diff-92dd155496a47c8983ad09bba2501210b1e822222469fd24381166a6f8d338f1R43)
when instantiate ne Builder you can pass method ref at object and when builder load data use this method to retrieve data and fill  field.

i've create data for test with  [Formio Docker](https://github.com/formio/formio)

File: formiodata/builder.py (hunk 1 above)
Comment (archetipo) [reply formiodata/builder.py:28]: Hi 

[here](https://github.com/novacode-nl/python-formio-data/pull/4/commits/84c64223d3bbdf894e305d1d956d6a36f885ff04#diff-92dd155496a47c8983ad09bba2501210b1e822222469fd24381166a6f8d338f1R44) pass the method ref

[here](https://github.com/novacode-nl/python-formio-data/pull/4/commits/84c64223d3bbdf894e305d1d956d6a36f885ff04#diff-85d655ecad9c721663f280667f10d84a4669d775f771a50d9c0e9be92c580e3eR348) if a condition is valid call method to retrieve datas and make the data list

Method can execute, external request, load file, o return a list of data.

**Comment by bobslee** [comment]
Hi @archetipo I just pushed a new change (release 0.2.0) which provides a FormRenderer class.
This powers nested (deep) Form components from submission data, loaded in the Form(Renderer) objet hierarchy.
Maybe you also need this for your project(s).

Sorry, unfortunately your PR now conflicts due to my changes. Please could you fix this, so I'm able to review and merge.

**Comment by bobslee** [comment]
@archetipo Ok now I understand this awesome feature :)
Would it be possible (and ideal) to provide a `resource_ext` function per component type or key?
So we're able to distinguish and implement `resource_ext` per component.
Maybe by turning this `kwarg` expecting a Dictionary (see below).

E.g in your unittest `tests/test_component_resource.py`

```python
resource_ext = {
  'selectone_places': self.places_resoure,
  'selectmulti_abc': self.abc_resource
}
self.builder_res_ext = Builder(self.builder_with_resource, resources_ext=resource_ext)
```

**Comment by bobslee** [comment]
Closed. No activity.

**Pull request closed without merging**